### PR TITLE
Optimizes queue_icon_update, reducing load times drastically

### PR DESCRIPTION
**Pull request by Andrew-Fall**
Changing from an associative list to two indexed lists cuts load times on large maps down drastically.
Also repaths SSicon_update from a processing subsystem, as it doesn't use the default /fire
## Changelog
:cl:
tweak: Drastically increases large map loading speed
/:cl:

### CODE REVIEW

File: code/controllers/subsystems/icon_updates.dm [hunk 1]
```diff
@@ -0,0 +1,71 @@
+SUBSYSTEM_DEF(icon_update)
+	name = "Icon Updates"
+	wait = 1	// ticks
+	flags = SS_TICKER
+	priority = SS_PRIORITY_ICON_UPDATE
+	init_order = SS_INIT_ICON_UPDATE
+
+	// Linked lists, queue_refs[x] should have TRUE or args stored in queue_args[x]
+	var/list/queue_refs = list()	// Atoms
+	var/list/queue_args = list()	// TRUE or args
```
Comment (Lohikar) [inline code/controllers/subsystems/icon_updates.dm]: This contains `null` if no args, not TRUE.

File: code/controllers/subsystems/icon_updates.dm [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+/datum/controller/subsystem/icon_update/stat_entry()
+	..("Queue Refs: [queue_refs.len], Queue Args: [queue_args.len]")
```
Comment (Lohikar) [inline code/controllers/subsystems/icon_updates.dm]: These two lists should always be the same length so just something like `Q:[queue_refs.len]` would be enough.

File: code/controllers/subsystems/icon_updates.dm [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	
+		if(Master.map_loading)
+			return
+
+		// Pops the atom and it's args, while making sure
+		// to keep them in sync in case runtimes/badmin
+		var/atom/A = queue_refs[queue_refs.len]
+		var/myArgs = queue_args[queue_args.len]
+
+		queue_args.len = --queue_refs.len
```
Comment (Lohikar) [inline code/controllers/subsystems/icon_updates.dm]: Ew, pls no:
```
queue_args.len -=1
queue_refs.len -= 1
```

File: code/controllers/subsystems/icon_updates.dm [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+			return
+
+/atom
+	var/icon_update_queued = FALSE
+
+/atom/proc/queue_icon_update(...)
+	// Skips if this is already queued
+	if(!icon_update_queued)
+
+		icon_update_queued = TRUE
```
Comment (Lohikar) [inline code/controllers/subsystems/icon_updates.dm:56]: This is never unset.

File: code/controllers/subsystems/icon_updates.dm [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+		SSicon_update.queue_refs.Add(src)
+
+		// Makes sure these are in sync, in case runtimes/badmin
```
Comment (Lohikar) [inline code/controllers/subsystems/icon_updates.dm:59]: If they're not the same length, the list is invalid anyways.

Comment (Andrew-Fall) [reply code/controllers/subsystems/icon_updates.dm:59]: Lowers the risk, and doesn't really change functionality, as queue_args len needs to be increased anyways. Looks clean too

**Review by MistakeNot4892 — DISMISSED** [review]

**Review by Lohikar — APPROVED** [review]

File: code/controllers/subsystems/icon_updates.dm [hunk 6]
```diff
@@ -0,0 +1,73 @@
+SUBSYSTEM_DEF(icon_update)
+	name = "Icon Updates"
+	wait = 1	// ticks
+	flags = SS_TICKER
```
Comment (Lohikar) [inline code/controllers/subsystems/icon_updates.dm]: This shouldn't be a ticker. Remove this flag please.
(You'll need to also move the define in subsystem-priority.dm)

**Review by Lohikar — CHANGES_REQUESTED** [review]

**Review by MistakeNot4892 — DISMISSED** [review]

**Review by Lohikar — APPROVED** [review]
Looks good, though this should be squash-merged.

**Pull request merged**